### PR TITLE
goto_trace: unbounded integers are now shown like other numbers

### PR DESCRIPTION
**Pull request by kroening**
- [ ] Each commit message has a non-empty body, explaining why the change was made.
- n/a Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- n/a My commit message includes data points confirming performance improvements (if claimed).
- [X] My PR is restricted to a single feature or bugfix.
- [X] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

**Review by allredj — COMMENTED** [review]
:no_entry_sign:
This PR failed Diffblue compatibility checks (cbmc commit: f73969caf70c477da8798def090e0ab3923e1871).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/90165384
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).

The incompatibility may have been introduced by an earlier PR. In that case merging this
PR should be avoided unless it fixes the current incompatibility.

**Review by tautschnig — APPROVED** [review]
Do the SMT-solver backed tests exercise this bit of code? If so, could an explicit test be added?

**Comment by kroening** [comment]
We can add a test either if
a) we activate the SMT-backend tests and have a solver that supports unbounded integers; or
b) we add (at least basic) support for unbounded integers to the internal solver.

**Comment by tautschnig** [comment]
We already have `regression/cbmc/integer-assignments1`, maybe that should include some patterns to inspect the trace output?

**Comment by kroening** [comment]
Indeed, done

File: regression/cbmc/integer-assignments1/test.desc [hunk 1]
```diff
@@ -3,5 +3,5 @@ main.c
 --trace --smt2
 ^EXIT=10$
 ^SIGNAL=0$
-^  b=100
+^  b=100 (0b1100100)$
```
Comment (tautschnig) [inline regression/cbmc/integer-assignments1/test.desc]: The parentheses will need escaping. (CI currently passes as we don't have SMT in there yet.)

**Review by tautschnig — CHANGES_REQUESTED** [review]

**Review by allredj — COMMENTED** [review]
:no_entry_sign:
This PR failed Diffblue compatibility checks (cbmc commit: a4462eb67f15ab4acc02f6baf30b98772ee0fc4c).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/90246246
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).

The incompatibility may have been introduced by an earlier PR. In that case merging this
PR should be avoided unless it fixes the current incompatibility.

File: regression/cbmc/integer-assignments1/test.desc (hunk 1 above)
Comment (kroening) [reply regression/cbmc/integer-assignments1/test.desc]: done

**Review by tautschnig — APPROVED** [review]

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 4e87b90625feec0698297dd4622ca23698fb03bb).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/90264632

**Pull request merged**